drivers/bridge/port_mapping.go

```diff
@@ -25,11 +26,23 @@ func (n *bridgeNetwork) allocatePorts(ep *bridgeEndpoint, reqDefBindIP net.IP, u
 		defHostIP = reqDefBindIP
 	}
 
-	return n.allocatePortsInternal(ep.extConnConfig.PortBindings, ep.addr.IP, defHostIP, ulPxyEnabled)
+	var pb []types.PortBinding
+
+	if ep.addrv6 != nil {
+		pb, _ = n.allocatePortsInternal(ep.extConnConfig.PortBindings, ep.addrv6.IP, defaultBindingIPV6, ulPxyEnabled, nil)
+	}
+
+	return n.allocatePortsInternal(ep.extConnConfig.PortBindings, ep.addr.IP, defHostIP, ulPxyEnabled, pb)
```

Comment:
above comment will ensure we don't change the signature of this API for now

Reply:
I have reworked this section a bit and also handle the `EnableIP6Tables` correctly.
I also changed the signature back and handle the different calls completely outside.
Hope this is a proper way to handle the port allocation.